cmd/minikube/cmd/start_test.go

```diff
@@ -477,6 +477,38 @@ func TestValidateRuntime(t *testing.T) {
 	}
 }
 
+func TestValidateWindowsOSVersion(t *testing.T) {
+	var tests = []struct {
+		osVersion string
+		errorMsg  string
+	}{
+		{
+			osVersion: "2019",
+			errorMsg:  "",
+		},
+		{
+			osVersion: "2022",
+			errorMsg:  "",
+		},
+		{
+			osVersion: "2023",
+			errorMsg:  "Invalid Windows Server OS Version: 2023. Valid OS version are: [2019 2022]",
```

Comment:
The failure comparison will be much simpler with `errors.New`, then you can compare directly check if `got != test.wantedError`

Reply:
I did realize that `errors.New` creates a new error value each time it's called so the comparison fails and 

How does this alternative implementation look? 

```
func TestValidateWindowsOSVersion(t *testing.T) {
	var tests = []struct {
		osVersion   string
		wantedError error
	}{
		{
			osVersion:   "2019",
			wantedError: nil,
		},
		{
			osVersion:   "2022",
			wantedError: nil,
		},
		{
			osVersion:   "2023",
			wantedError: errors.New("Invalid Windows Server OS Version: 2023. Valid OS version are: [2019 2022]"),
		},
	}
	for _, test := range tests {
		t.Run(test.osVersion, func(t *testing.T) {
			got := validateWindowsOSVersion(test.osVersion)
			if got != nil && test.wantedError != nil {
				if got.Error() != test.wantedError.Error() {
					t.Errorf("ValidateWindowsOSVersion(osVersion=%s): got %v, expected %v", test.osVersion, got, test.wantedError)
				}
			} else if got != test.wantedError {
				t.Errorf("ValidateWindowsOSVersion(osVersion=%s): got %v, expected %v", test.osVersion, got, test.wantedError)
			}
		})
	}
}
```

The first `if` block checks if both `got `and `wantedError `are not `nill`. If true, it then compares the error messgaes of `got `and `wantedError `using their `Error()` methods. 
The `else if` block handles cases where the expectations about the error presence differ: either `got `or `wantedError `is nil, but not both, or both are `nil`.

This  passes the test!

Reply:
Sorry, I'm used to us wrapping errors, I see that just creating a new error doesn't work. What you have is good then